apps/emqx/src/emqx_ws_connection.erl

```diff
@@ -1013,6 +1013,10 @@ classify([Cmd = {shutdown, _Reason} | More], Packets, Cmds, Events) ->
     classify(More, Packets, [Cmd | Cmds], Events);
 classify([Cmd = close | More], Packets, Cmds, Events) ->
     classify(More, Packets, [Cmd | Cmds], Events);
+%% cowboy_websocket's close reason must be an atom to avoid crashing the sender process.
+%% The cause reasons come from parse_frame_error.
```

Comment:
do we emit string format `cause`? maybe change all to string.

Reply:
Yes, all of them are atom causes
 
from here： https://github.com/emqx/emqx/commit/8b0e15e4020760cff017a74e8431c608ac8643f2